portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,16 +38,22 @@
   private Gson gson = new Gson();
   private static final String COMMENT_KIND = "Comment";
   private static final String TEXT_KEY = "text";
+  private static final String TIMESTAMP_KIND = "timestamp";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query(COMMENT_KIND);
+    Query query = new Query(COMMENT_KIND).addSort(TIMESTAMP_KIND, SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
+    int numComments = getNumberOfComments(request, "comments-number");
 
     comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
+    Iterator commentsIterator = results.asIterator();
+
+    // Accept either all comments or as many as specified in numComments.
+    for (int i = 0; i < numComments && commentsIterator.hasNext(); i++) {
```

Comment:
Using the Guava iterables class would be a cleaner implementation for this. (iterables.limit)

https://guava.dev/releases/20.0/api/docs/com/google/common/collect/Iterables.html